sqlglot/parser.py

```diff
@@ -1010,6 +1058,16 @@ def _parse_create(self) -> t.Optional[exp.Expression]:
             begin=begin,
         )
 
+    def _parse_property_before(self) -> t.Optional[exp.Expression]:
+        self._match_text_seq("NO")
```

Comment:
what's the point of this? doesn't this just ignore no, dual, and default?

Reply:
no/dual/default can be the first word of multiple options, so they have to be consumed before we can figure out which option parser to call.

These do that consumption, then the option parsers determine no/dual/default by looking at the previous token (like "BEFORE" on line 567).

One alternative approach would be to do the no/dual/default routing here like:
```
if self._match_text_seq("NO"):
    if self.PROPERTY_PARSERS_TEXT.get(self._curr.text.upper()):
        return self.PROPERTY_PARSERS_TEXT[self._curr.text.upper()](self, no=True)
```

Reply:
The parsers identify no/dual/default by looking at _prev, so we need those to have already been consumed.